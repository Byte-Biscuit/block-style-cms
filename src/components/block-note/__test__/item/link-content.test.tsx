import React from "react";
import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import FormattedLink from "../../item/link-content";
import type { LinkContent } from "../../meta";

describe("FormattedLink", () => {
    it("renders link with nested text content and applies correct styles", () => {
        const item = {
            type: "link",
            href: "https://studio.xiyue.space",
            content: [
                {
                    type: "text",
                    text: "文",
                    styles: {},
                },
                {
                    type: "text",
                    text: "字",
                    styles: {
                        bold: true,
                        italic: true,
                        underline: true,
                        strike: true,
                        textColor: "red",
                    },
                },
            ],
        } as LinkContent;

        const { asFragment } = render(<FormattedLink item={item} />);

        // 验证链接存在且href正确
        const linkEl = screen.getByRole("link") as HTMLAnchorElement;
        expect(linkEl).toBeTruthy();
        expect(linkEl.href).toBe("https://studio.xiyue.space/");

        // 验证第一个文本节点 "文" (无样式)
        const firstTextEl = screen.getByText("文");
        expect(firstTextEl).toBeTruthy();

        // 验证第二个文本节点 "字" (有样式)
        const secondTextEl = screen.getByText("字");
        expect(secondTextEl).toBeTruthy();
        expect(secondTextEl.tagName).toBe("SPAN");

        const secondClassName = secondTextEl.className;
        expect(secondClassName).toContain("font-bold");
        expect(secondClassName).toContain("italic");
        expect(secondClassName).toContain("underline");
        expect(secondClassName).toContain("line-through");
        expect(secondClassName).toContain("text-red-600");

        // snapshot 验证
        expect(asFragment()).toMatchSnapshot();
    });
});
