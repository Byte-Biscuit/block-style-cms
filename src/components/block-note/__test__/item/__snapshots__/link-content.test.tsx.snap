// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormattedLink > renders link with nested text content and applies correct styles 1`] = `
<DocumentFragment>
  <a
    class="break-words"
    href="https://studio.xiyue.space"
    rel="noopener noreferrer"
    target="_blank"
  >
    文
    <span
      class="font-bold italic underline line-through text-red-600"
    >
      字
    </span>
  </a>
</DocumentFragment>
`;
